File: sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/view/UpdateContext.java
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.javasdk.view;

import kalix.javasdk.CloudEvent;
import kalix.javasdk.MetadataContext;

import java.util.Optional;

/** Context for view update calls. */
public interface UpdateContext extends ViewContext, MetadataContext {

  /**
   * The origin subject of the {@link CloudEvent}. For example, the entity id when the event was
   * emitted from an entity.
   */
  Optional<String> eventSubject();

  /** The name of the event being handled. */
  String eventName();
}

File: sdk/scala-sdk-protobuf/src/main/scala/kalix/scalasdk/view/UpdateContext.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.scalasdk.view

import kalix.scalasdk.MetadataContext

/** Context for view update calls. */
trait UpdateContext extends ViewContext with MetadataContext {

  /**
   * The origin subject of the [[kalix.scalasdk.CloudEvent]]. For example, the entity id when the event was emitted from
   * an entity.
   */
  def eventSubject: Option[String]

  /** The name of the event being handled. */
  def eventName: String
}
